[BUG] Unable to create Policy violations from namespace controller

Engine generate does not return patched resource, as its not needed but this is needed to create pv owners.

After adding a resource, the  controller panics:
```
panic: interface conversion: interface {} is *v1alpha1.ClusterPolicyViolation, not *v1alpha1.NamespacedPolicyViolation [recovered]
	panic: interface conversion: interface {} is *v1alpha1.ClusterPolicyViolation, not *v1alpha1.NamespacedPolicyViolation

goroutine 428 [running]:
github.com/nirmata/kyverno/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/Users/shivd/go/src/github.com/nirmata/kyverno/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:58 +0x105
panic(0x1f5a060, 0xc0005780c0)
	/usr/local/Cellar/go/1.12.9/libexec/src/runtime/panic.go:522 +0x1b5
github.com/nirmata/kyverno/pkg/policy.(*PolicyController).addNamespacedPolicyViolation(0xc00018c360, 0x20a4040, 0xc000104d00)
	/Users/shivd/go/src/github.com/nirmata/kyverno/pkg/policy/namespacedpv.go:13 +0x5d8
github.com/nirmata/kyverno/vendor/k8s.io/client-go/tools/cache.ResourceEventHandlerFuncs.OnAdd(...)

``

--- a/pkg/engine/generation.go
+++ b/pkg/engine/generation.go
@@ -44,6 +44,8 @@ func Generate(client *client.Client, policy kyverno.ClusterPolicy, ns unstructur
 		response.PolicyResponse.Rules = append(response.PolicyResponse.Rules, ruleResponse)
 		incrementAppliedRuleCount()
 	}
+	// set resource in reponse
+	response.PatchedResource = ns
 	return response
 }
 

--- a/pkg/engine/validation.go
+++ b/pkg/engine/validation.go
@@ -74,6 +74,9 @@ func Validate(policyContext PolicyContext) (response EngineResponse) {
 		// there are changes send response
 		startResultResponse(newResponse, policy, newR)
 		defer endResultResponse(newResponse, startTime)
+		if reflect.DeepEqual(newResponse.PatchedResource, unstructured.Unstructured{}) {
+			newResponse.PatchedResource = newR
+		}
 		return *newResponse
 	}
 	// if there are no changes with old and new response then sent empty response

--- a/pkg/policy/controller.go
+++ b/pkg/policy/controller.go
@@ -131,7 +131,7 @@ func NewPolicyController(kyvernoClient *kyvernoclient.Clientset, client *client.
 		DeleteFunc: pc.deletePolicyViolation,
 	})
 
-	pvInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+	nspvInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
 		AddFunc:    pc.addNamespacedPolicyViolation,
 		UpdateFunc: pc.updateNamespacedPolicyViolation,
 		DeleteFunc: pc.deleteNamespacedPolicyViolation,
@@ -146,7 +146,7 @@ func NewPolicyController(kyvernoClient *kyvernoclient.Clientset, client *client.
 
 	pc.pListerSynced = pInformer.Informer().HasSynced
 	pc.pvListerSynced = pvInformer.Informer().HasSynced
-	pc.nspvListerSynced = pvInformer.Informer().HasSynced
+	pc.nspvListerSynced = nspvInformer.Informer().HasSynced
 
 	pc.mutationwebhookLister = webhookInformer.Lister()
 
@@ -394,7 +394,7 @@ func (pc *PolicyController) Run(workers int, stopCh <-chan struct{}) {
 	glog.Info("Starting policy controller")
 	defer glog.Info("Shutting down policy controller")
 
-	if !cache.WaitForCacheSync(stopCh, pc.pListerSynced, pc.pvListerSynced) {
+	if !cache.WaitForCacheSync(stopCh, pc.pListerSynced, pc.pvListerSynced, pc.nspvListerSynced) {
 		return
 	}
 	for i := 0; i < workers; i++ {

